src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/ExternalSettingsProviderService.cs

```diff
@@ -0,0 +1,229 @@
+        private const string MonikerShowPackageManagementChooser = "packageManagement.showPackageManagementChooser";
+
+        private readonly ISettings _settings;
+        private readonly VSSettings _vsSettings;
+
+        private PackageRestoreConsent _packageRestoreConsent;
+        private BindingRedirectBehavior _bindingRedirectBehavior;
+        private PackageManagementFormat _packageManagementFormat;
+
+        public ExternalSettingsProviderService()
+        {
+            var componentModel = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            _settings = componentModel.GetService<ISettings>();
+            _vsSettings = _settings as VSSettings;
+            if (_vsSettings != null)
+            {
+                _vsSettings.SettingsChanged += VsSettings_SettingsChanged;
+            }
+            Debug.Assert(_settings != null);
+        }
+
+        private void VsSettings_SettingsChanged(object sender, EventArgs e)
+        {
+            _packageRestoreConsent = null;
+            _bindingRedirectBehavior = null;
+            _packageManagementFormat = null;
+            SettingValuesChanged.Invoke(this, ExternalSettingsChangedEventArgs.SomeOrAll);
+        }
+
+        private BindingRedirectBehavior BindingRedirectBehavior
+        {
+            get
+            {
+                if (_bindingRedirectBehavior is null)
+                {
+                    _bindingRedirectBehavior = new BindingRedirectBehavior(_settings);
+                }
+
+                return _bindingRedirectBehavior;
+            }
+        }
+
+        private PackageRestoreConsent PackageRestoreConsent
+        {
+            get
+            {
+                if (_packageRestoreConsent is null)
+                {
+                    _packageRestoreConsent = new PackageManagement.PackageRestoreConsent(_settings);
+                }
+
+                return _packageRestoreConsent;
+            }
+        }
+
+        private PackageManagementFormat PackageManagementFormat
+        {
+            get
+            {
+                if (_packageManagementFormat is null)
+                {
+                    _packageManagementFormat = new PackageManagementFormat(_settings);
+                }
+
+                return _packageManagementFormat;
+            }
+        }
+
+        public event EventHandler<ExternalSettingsChangedEventArgs> SettingValuesChanged;
+        public event EventHandler<EnumSettingChoicesChangedEventArgs> EnumSettingChoicesChanged { add { } remove { } }
+        public event EventHandler<DynamicMessageTextChangedEventArgs> DynamicMessageTextChanged { add { } remove { } }
+        public event EventHandler ErrorConditionResolved { add { } remove { } }
+
+        public void Dispose()
+        {
+            if (_vsSettings != null)
+            {
+                _vsSettings.SettingsChanged -= VsSettings_SettingsChanged;
+            }
```

Comment:
nit
```suggestion
        _vsSettings?.SettingsChanged -= VsSettings_SettingsChanged;
```

Reply:
This causes https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0070?f1url=%3FappId%3Droslyn%26k%3Dk(CS0070)
![image](https://github.com/user-attachments/assets/1ca7cb73-a463-490d-86b1-80415d2c25bd)

Reply:
Oops, sorry I was thinking of `class.Event?.Invoke(sender, args)`